src/spikeinterface/preprocessing/astype.py

```diff
@@ -0,0 +1,52 @@
+from ..core.core_tools import define_function_from_class
+from .basepreprocessor import BasePreprocessor, BasePreprocessorSegment
+from .filter import fix_dtype
+
+
+class AstypeRecording(BasePreprocessor):
+    """The spikeinterface analog of numpy.astype
+
+    Converts a recording to another dtype on the fly.
+    """
+
+    name = "astype"
+
+    def __init__(
+        self,
+        recording,
+        dtype=None,
+    ):
+        dtype_ = fix_dtype(recording, dtype)
+        BasePreprocessor.__init__(self, recording, dtype=dtype_)
+
+        for parent_segment in recording._recording_segments:
+            rec_segment = AstypeRecordingSegment(
+                parent_segment,
+                dtype,
+            )
+            self.add_recording_segment(rec_segment)
+
+        self._kwargs = dict(
+            recording=recording,
+            dtype=dtype_.str,
+        )
+
+
+class AstypeRecordingSegment(BasePreprocessorSegment):
+    def __init__(
+        self,
+        parent_recording_segment,
+        dtype,
+    ):
+        BasePreprocessorSegment.__init__(self, parent_recording_segment)
+        self.dtype = dtype
+
+    def get_traces(self, start_frame, end_frame, channel_indices):
+        if channel_indices is None:
+            channel_indices = slice(None)
+        traces = self.parent_recording_segment.get_traces(start_frame, end_frame, channel_indices)
+        return traces.astype(self.dtype)
```

Comment:
I am flagging thsi as a point of discssuion. When I was writing on fixin the memory leak of writing I realized that this function always creates a malloc (even if the types) are the same:

https://github.com/SpikeInterface/spikeinterface/pull/1602/files#r1204822550

I am not sure what behavior you guys want here (maybe always returning a copy is the desired behavior) but I wanted to let you know of this in case is important:

https://numpy.org/doc/stable/reference/generated/numpy.ndarray.astype.html

Reply:
I never knew! I guess we can set `copy=False` to fix that. Will do

Reply:
I think there is some other casting down there to float32 that can be avoided.

Reply:
Yeah, it is terrible! Anyway, copy=False has been added in the cases here. Thanks @h-mayorquin !